website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,799 @@
+                                args.data().map(function(item, index, array){
+                                    if(index === array.length-1){
+                                        return m('.fb-parent-row.btn', {
+                                            style : 'margin-left:' + (index*20) + 'px;',
+                                        },  [
+                                            m('i.fa.fa-angle-right.m-r-xs'),
+                                            item.label
+                                        ]);
+                                    }
+                                    item.index = index; // Add index to update breadcrumbs
+                                    item.placement = 'breadcrumb'; // differentiate location for proper breadcrumb actions
+                                    return m('.fb-parent-row',
+                                        m('span.btn.btn-link', {
+                                            style : 'margin-left:' + (index*20) + 'px;',
+                                            onclick : function() {
+                                                args.updateFilesData(item);
+                                                $('.modal').modal('hide');
+                                            }
+                                        },  [
+                                            m('i.fa.fa-angle-right.m-r-xs'),
+                                            item.label
+                                        ])
+                                    );
+                                })
+                            ]),
+                        ])
+                    )
+                )
+            ]);
+        }
+        return m('.fb-breadcrumbs', m('ul', [
+            args.data().map(function(item, index, array){
+                if(index === array.length-1){
+                    return m('li',  m('span.btn', item.label));
+                }
+                item.index = index; // Add index to update breadcrumbs
+                item.placement = 'breadcrumb'; // differentiate location for proper breadcrumb actions
+                return m('li',
+                    m('span.btn.btn-link', { onclick : args.updateFilesData.bind(null, item)},  item.label),
+                    m('i.fa.fa-angle-right')
+                );
+            })
+
+        ]));
+    }
+};
+
+
+/**
+ * Filters Module.
+ * @constructor
+ */
+var Filters = {
+    view : function (ctrl, args) {
+        var selectedCSS;
+        return m('.fb-filters.m-t-lg',
+            [
+                m('h5', 'Contributors'),
+                m('ul', [
+                    args.nameFilters.map(function(item, index){
+                        selectedCSS = item.id === args.activeFilter() ? '.active' : '';
+                        return m('li' + selectedCSS,
+                            m('a', { href : '#', onclick : args.updateFilter.bind(null, item)}, item.label)
```

Comment:
I was trying to bind a function (i.e. not call) without binding this to it, but since the function uses self for its main context it doesn't really matter. I can pass this but it won't be used necessarily. What's the best thing to do here?